core/domain/rights_manager.py

```diff
@@ -877,21 +754,24 @@ def _assign_role(
         Exception. The activity is already publicly viewable.
         Exception. The role is invalid.
     """
-    if not Actor(committer_id).can_modify_roles(activity_type, activity_id):
+    committer_role = user_services.get_user_role_from_id(committer_id)
+    user = user_services.UserActionsInfo(committer_id, committer_role)
+    activity_rights = _get_activity_rights(activity_type, activity_id)
```

Comment:
It is quite amusing to me that these three completely different lines align so neatly at the right edge :P